cmd/fabric8-jenkins-proxy/main_cors_test.go

```diff
@@ -31,37 +27,54 @@ func TestAPIServerCORSHeaders(t *testing.T) {
 
 	reader, _ := http.NewRequest("POST", "/doesntmatter", nil)
 
-	// Check for origin "https://*.openshift.io"
+	// hope UUID generates valid characters for domain name
 	randomOrigin := uuid.NewV4().String()
-	reader.Header.Set("Origin", "https://"+randomOrigin+".openshift.io")
-	writer := httptest.NewRecorder()
-	apiServer.Handler.ServeHTTP(writer, reader)
-	assert.Equal(t, "https://"+randomOrigin+".openshift.io", writer.Header().Get("access-control-allow-origin"))
 
-	// Check for origin "https://openshift.io"
-	reader.Header.Set("Origin", "https://openshift.io")
-	writer = httptest.NewRecorder()
-	apiServer.Handler.ServeHTTP(writer, reader)
-	assert.Equal(t, "https://openshift.io", writer.Header().Get("access-control-allow-origin"))
+	t.Run("Check for origin https://*.openshift.io", func(t *testing.T) {
```

Comment:
+1 to using t.Run but create a test table `tt` and  iterate `tt` to run each test 
This video gives you a good intro to go test - https://www.youtube.com/watch?v=hVFEV-ieeew

Reply:
thanks for link!
Could have gone for `tt` but the side effect is, it hampers the readability. _The test case should be easy to understand and read_.
Still shall we go for test tabel?

Reply:
> hampers readabilty 
Well think of it as a test function name and not a description so that you can run just a single test (a test case in `tt`) using `go test -run `

Reply:
Yes, a meaningful name is better than description. I mean test table is a good approach, only thing is, other developers may need to understand the test logic as well :wink:.

_added `tt`_